File: solutions/leetcode/contests/w-150/1-sum_of_good_numbers.java
class Solution {
    public int sumOfGoodNumbers(int[] nums, int k) {
        int sum = 0;
        int n = nums.length;

        for (int i = 0; i < n; i++) {
            boolean ans = true;
            if ((i - k >= 0 && nums[i] <= nums[i - k]) || (i + k < n && nums[i] <= nums[i + k])) ans = false;
            if (ans) sum += nums[i];
        }
        return sum;
    }
}

File: solutions/leetcode/contests/w-150/1-sum_of_good_numbers.scala
object Solution {
    def sumOfGoodNumbers(arr: Array[Int], gap: Int): Int = {
        var total = 0
        val length = arr.length

        for (i <- arr.indices) {
            var valid = true

            if (i - gap >= 0 && arr(i) <= arr(i - gap)) valid = false
            if (i + gap < length && arr(i) <= arr(i + gap)) valid = false

            if (valid) total += arr(i)
        }

        total
    }
}

